K2Bridge/Visitors/RangeVisitor.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Globalization;
+    using K2Bridge.Models.Request.Aggregations;
+    using K2Bridge.Models.Response;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(RangeAggregation rangeAggregation)
+        {
+            Ensure.IsNotNull(rangeAggregation, nameof(RangeAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            // Start the union operator
+            rangeAggregation.KustoQL += "union ";
+
+            // Query expressions for each range
+            // (_data | where foo >= 1 and bar < 10 | summarize ['3']=avg(baz), count() | extend ['2'] = '1-10')
+            //          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            //          The range clause
+            //                                        ^^^^^^^^^^^^^^^^^^^^^^^^
+            //                                        The metrics expressions
+            //                                                                  ^^^^^^^
+            //                                                                  Default metric count()
+            //                                                                            ^^^^^^^^^^^^^^^^^^^^^
+            //                                                                            Column with the range name
+            foreach (var range in rangeAggregation.Ranges)
+            {
+                rangeAggregation.KustoQL += $"(_data | {KustoQLOperators.Where} ";
+
+                var rangeClause = new Models.Request.Queries.RangeClause()
+                {
+                    FieldName = rangeAggregation.Field,
+                    GTEValue = range.From?.ToString() ?? "*",
+                    LTValue = range.To?.ToString() ?? "*",
+                };
```

Comment:
I think this section should happen at the json to models conversion part.

because, line 40, where you create the clause is not the natural place to create a new clause object. wdyt?

Reply:
Well the idea was to reuse the RangeClause to assemble part of the query, here in the RangeVisitor, since it happens to produce the appropriate KustoQL... I guess we could assemble the RangeClause at conversion time, but right now there is no custom JSON converter for the RangeAggregation, since we are trying to simplify that step. The model is straight generated from incoming JSON.

Reply:
I understand, but still. we can only simplify so much. and it seems we moved code of the convertor to the visitor just because we want to keep the convertor simple (or non existent)..

Reply:
OK, I guess this is a design decision we should discuss with the team to make sure we are aligned. Indeed, I think we have been moving some logic out of the converters, to centralize it in the visitors.

Reply:
lets discuss today

Reply:
For me, the serialization is good. We populate the RangeAggregationExpression collection with the correct fields. Then we have a decision later in the visitor 1/ Should we convert RangeAggregationExpression to RangeClause to reuse the visitor associated with -or 2/ Should we create a dedicated visitor for  RangeAggregationExpression.

Reply:
I think Elad's point is: the Model should not necessarily represent the JSON, it should represent whatever the Visitor needs, and it is the conversion layer's job to assemble this Model. So, for example, here, we could have a list of RangeClause in the model instead of RangeAggregationExpression, and a custom converter to create it. In this way the Visitor does not have to care about conversion.

Reply:
As discussed, just pushed the `RangeAggregationExpression` visitor.